src/WebJobs.Script.WebHost/WebJobsApplicationBuilderExtension.cs

```diff
@@ -59,7 +60,10 @@ public static IApplicationBuilder UseWebJobsScriptHost(this IApplicationBuilder
                 config.UseMiddleware<HttpThrottleMiddleware>();
             });
             builder.UseMiddleware<ResponseContextItemsCheckMiddleware>();
-            builder.UseMiddleware<JobHostPipelineMiddleware>();
+            builder.UseWhen(context => context.RequestServices.GetService<IJobHostMiddlewarePipeline>() != null, config =>
```

Comment:
This has been addressed with #4502 , so this can be removed. With the approach in that PR, this check isn't necessary.